ext/mruby_engine/dlmalloc.c

```diff
@@ -4523,6 +4527,40 @@ struct mallinfo mspace_mallinfo(mspace msp) {
   }
   return internal_mallinfo(ms);
 }
+
+void log_stuff(mspace msp) {
+  mstate m = (mstate)msp;
+  ensure_initialization();
+  if (!PREACTION(m)) {
+    check_malloc_state(m);
+    if (is_initialized(m)) {
+      size_t nfree = SIZE_T_ONE; /* top always free */
+      size_t mfree = m->topsize + TOP_FOOT_SIZE;
+      size_t sum = mfree;
+      msegmentptr s = &m->seg;
+      while (s != 0) {
+        fprintf(stderr, "[SCRIPT-DEBUG] S @ %p\n", (void *)s);
+        mchunkptr q = align_as_chunk(s->base);
+        while (segment_holds(s, q) &&
+               q != m->top && q->head != FENCEPOST_HEAD) {
+          size_t sz = chunksize(q);
+          sum += sz;
+          if (!is_inuse(q)) {
+            mfree += sz;
+            ++nfree;
+          }
+          q = next_chunk(q);
+        }
+        s = s->next;
+      }
+
+      fprintf(stderr, "[SCRIPT-DEBUG] %zu, %zu, %zu, %zu, %zu, %zu, %zu, %zu\n",
```

Comment:
If you want Splunk to automatically extract a facet for you, you need to print something like `sum=%zu`.